datafusion/expr/src/expr.rs

```diff
@@ -99,6 +99,15 @@ pub enum Expr {
         /// Right-hand side of the expression
         right: Box<Expr>,
     },
+    /// A binary expression such as "age > 21"
+    AnyExpr {
+        /// Left-hand side of the expression
+        left: Box<Expr>,
+        /// The comparison operator
+        op: Operator,
+        /// Right-hand side of the expression
+        right: Box<Expr>,
```

Comment:
The `Expr` here could be an `Expr::ScalarSubquery` to cover the subquery case (assuming that we only need to support subqueries that select a single column?)

Reply:
The definition of `AnyExpr` here is identical to the definition of `BinaryExpr`. I wonder if it would be better to use `BinaryExpr` for consistency and model Any as `AnyExpr(Expr)` to just model the right-hand side?